src/hotspot/share/memory/arena.cpp

```diff
@@ -42,93 +42,81 @@ STATIC_ASSERT(is_aligned((int)Chunk::medium_size, ARENA_AMALLOC_ALIGNMENT));
 STATIC_ASSERT(is_aligned((int)Chunk::size, ARENA_AMALLOC_ALIGNMENT));
 STATIC_ASSERT(is_aligned((int)Chunk::non_pool_size, ARENA_AMALLOC_ALIGNMENT));
 
-//--------------------------------------------------------------------------------------
-// ChunkPool implementation
-
 // MT-safe pool of same-sized chunks to reduce malloc/free thrashing
 // NB: not using Mutex because pools are used before Threads are initialized
 class ChunkPool {
-  Chunk*       _first;        // first cached Chunk; its first word points to next chunk
-  const size_t _size;         // (inner payload) size of the chunks this pool serves
-
   // Our four static pools
   static const int _num_pools = 4;
   static ChunkPool _pools[_num_pools];
 
- public:
-  ChunkPool(size_t size) : _first(nullptr), _size(size) {}
+  Chunk*       _first;        // first cached Chunk; its first word points to next chunk
+  const size_t _size;         // (inner payload) size of the chunks this pool serves
 
   // Allocate a chunk from the pool; returns null if pool is empty.
-  Chunk* allocate() {
-    ThreadCritical tc;
-    Chunk* c = _first;
-    if (_first != nullptr) {
-      _first = _first->next();
-    }
-    return c;
-  }
-
+  Chunk* take_from_pool();
   // Return a chunk to the pool
-  void free(Chunk* chunk) {
-    assert(chunk->length() == _size, "wrong pool for this chunk");
-    ThreadCritical tc;
-    chunk->set_next(_first);
-    _first = chunk;
-  }
-
+  void return_to_pool(Chunk* chunk);
   // Prune the pool
-  void prune() {
-    // Free all chunks while in ThreadCritical lock
-    // so NMT adjustment is stable.
-    ThreadCritical tc;
-    Chunk* cur = _first;
-    Chunk* next = nullptr;
-    while (cur != nullptr) {
-      next = cur->next();
-      os::free(cur);
-      cur = next;
-    }
-    _first = nullptr;
-  }
-
-  static void clean() {
-    for (int i = 0; i < _num_pools; i++) {
-      _pools[i].prune();
-    }
-  }
+  void prune();
```

Comment:
Comment just repeats the name. I'd either omit it or describe what the function is doing.